pkg/host/store.go

```diff
@@ -129,6 +130,11 @@ func (s *Store[E]) Update(ctx context.Context, obj E) (E, error) {
 		return obj, nil
 	}
 
+	if oldObj.GetGeneration() != obj.GetGeneration() {
+		return utils.Zero[E](), fmt.Errorf("failed to update object: generation is not latest")
+	}
+	obj.SetGeneration(obj.GetGeneration() + 1)
```

Comment:
optional: i prefer create method IncrementGeneration() or create const for 1 because you call this code on multiple places.

Reply:
Done